tutorials/kotlin-springboot-container-engine.md

```diff
@@ -0,0 +1,383 @@
+
+    gradle bootRun
+
+
+**Note:** The `gradle bootRun` is a quick way to build and run the application. Later on when creating the Docker image, you'll 
+need to first build the app using the Gradle `build` task and then run it.
+
+6. Open the browser and make sure your get a valid JSON response when accessing
+
+    http://localhost:8080/message
+    
+The result should be
+
+```json
+{
+"text": "Hello from Google Cloud",
+"priority": "High"
+}
+```    
+    
+## Dockerizing your application
+
+The next step is to produce a Docker image that builds and runs your
+application in a Docker container. You will define this image using a
+Dockerfile.
+
+### Create a Dockerfile
+
+Various considerations go into designing a good Docker image. The Dockerfile
+used by this tutorial builds a release and runs it with Alpine Linux.
+If you are experienced with Docker, you can customize your image.
+
+1.  Create a file called `Dockerfile` in your project directory. Copy the
+    following content into it. Alternately, you can
+    [download](https://github.com/JetBrains/gcp-samples/blob/master/Dockerfile)
+    the Dockerfile to study and customize.
+
+        FROM openjdk:8-jdk-alpine
+        VOLUME /tmp
+        RUN mkdir /work
+        COPY . /work 
+        WORKDIR /work
+        RUN /work/gradlew build
+        RUN mv /work/build/libs/*.jar /work/app.jar
+        ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","/work/app.jar"]
+
+
+2.  Create a file called `.dockerignore` in your project directory. Copy the
+    following content into it. Alternately, you can
+    [download](https://github.com/JetBrains/gcp-samples/blob/master/.dockerignore)
+    the Dockerfile file to study and customize.
+
+        .gradle
+        build
+        out
+    
+
+The Dockerfile performs a series of steps. It creates a `work` folder where sources will be copied to and built from.
+It then copies the sources from the local file system (which should be your project folder) to the container. It sets the
+working directory to the `work` folder created and then runs the Gradle wrapper command (`gradlew`) to build the project. This wrapper
+automatically downloads Gradle and then builds the project. Once built, it then copies the output of the `jar` to the `work` folder
+ready for execution when invoking `docker run`.
+
+### Test the Dockerfile
+
+Build the image locally and test running your application from the image:
+
+    docker build --no-cache -t demo .
+    docker run -it --rm -p 8080:8080 demo
+
+The period at the end of the `docker build` command is required. It denotes the
+root directory of the application you are building.
+
+Visit [http://localhost:8080/message](http://localhost:8080/message) to see the application
+respond running locally from your Docker image.
+
+## Deploying your application
+
+Now you're ready to deploy your application to Kubernetes Engine!
+
+### Build the production image
+
+To deploy the app, you will use the
```

Comment:
To deploy the app, you will use [Container Builder](https://cloud.google.com/container-builder/) to build your Docker image in the cloud and store the resulting Docker image in your project in [Container Registry](https://cloud.google.com/container-registry/).